src/Tabs/Tabs.js

```diff
@@ -140,11 +141,23 @@ class Tabs extends Component {
   };
 
   componentDidMount() {
-    this.updatePositionStates(this.props);
+    this.updateIndicatorState(this.props);
+    this.updateScrollButtonState();
   }
 
   componentWillReceiveProps(nextProps) {
-    this.updatePositionStates(nextProps);
+    if (this.props.index !== nextProps.index) {
+      this.updateIndicatorState(nextProps);
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      this.props.width !== prevProps.width ||
+      !isEqual(this.state.indicatorStyle, prevState.indicatorStyle)
```

Comment:
> Can't we use strict comparison here?

@oliviertassinari I don't think so. `state.indicatorStyle` is an object with properties `left` and `width`.  `this.state` and `prevState` should be separate objects so AFAIK strict comparison would always return false.  `isEqual` is doing the deep comparison of the properties to get the truthy conditions.

It could be rewritten as:
```js
if (
  this.props.width !== prevProps.width ||
  this.state.indicatorStyle.left !== prevState.indicatorStyle.left ||
  this.state.indicatorStyle.width !== prevState.indicatorStyle.width
```
but I don't think that necessarily represents an improvement.